sway-lib-std/src/u256.sw

```diff
@@ -483,10 +483,19 @@ impl core::ops::Multiply for U256 {
                 let result_d_c = self.d.overflowing_mul(other.c);
                 let result_d_d = self.d.overflowing_mul(other.d);
 
+                let to_c_1 = result_d_d.upper.overflowing_add(result_c_d.lower);
```

Comment:
Right after the `else {` above, I would add a comment saying that in this branch, we know that `self.a`, `other.a`, self.b`, and `other.b` are all zeros.

Also, this is a bit confusing to read. I'm not sure the best way to comment this but walking through the process somehow could be really valuable.

Reply:
Done.

Changed naming, and added comprehensions, so hope it's more readable.